File: 2022.01.30/python/1874.py
n = int(input())
nums = [int(input()) for _ in range(n)]
stack = []
idx = 0
result = []
for i in range(1,n+1):
    stack.append(i)
    result.append("+")
    while stack and stack[-1] == nums[idx]:
        idx+=1
        result.append("-")
        stack.pop()
if stack: result = "NO"
else : result = "\n".join(map(str,result))
print(result)

File: Data_Structure/javascript/1874.js
const nums = require('fs').readFileSync('/dev/stdin').toString().split('\n').map(Number);
let stack = [];
let targetIdx = 1;
let result = [];
for (let i = 1 ; i <= nums[0] ; i++){
  stack.push(i);
  result.push("+");
  while (stack.length> 0 && stack[stack.length-1] === nums[targetIdx]) {
    stack.pop();
    result.push("-");
    targetIdx++;
  }
}
if (stack.length>0) result = "NO";
else result = result.join("\n");
console.log(result);